dfc/proxy.go

```diff
@@ -1839,18 +1845,21 @@ func (p *proxyrunner) httpcludel(w http.ResponseWriter, r *http.Request) {
 	}
 	p.smap.lock()
 	defer p.smap.unlock()
-	if !proxy && p.smap.get(sid) == nil {
+	si := p.smap.get(sid)
+	if !proxy && si == nil {
 		glog.Errorf("Unknown target %s", sid)
 		return
 	}
-	if proxy && p.smap.getProxy(sid) == nil {
+	if proxy && si == nil {
```

Comment:
smap methods get/del/add are a bit misleading. They work only with Tmap part, while getProxy/delProxy/addProxy work only with Pmap part. So, the patch does invalid check here.

We may think of adding 'Target' to 'add/del/etc' to avoid confusion in the future.